src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiRequestBodyGenerator.cs

```diff
@@ -66,15 +66,17 @@ public static OpenApiRequestBody CreateRequestBody(this ODataContext context, IE
                 Properties = new Dictionary<string, OpenApiSchema>()
             };
 
+            bool allParamsNullable = true;
             foreach (var parameter in action.Parameters.Skip(skip))
             {
+                allParamsNullable &= parameter.Type.IsNullable;
```

Comment:
One last question here. Should we be checking if the parameter is of type `IEdmOptionalParameter` instead? Is it possible for required parameter to be nullable?

Reply:
Yes indeed! We should. I will add a check for the annotation: `Org.OData.Core.V1.OptionalParameter`

Reply:
I think for now let's only update the check to only cater for the annotation `Org.OData.Core.V1.OptionalParameter` as that's the scenario we want to cater for in https://github.com/microsoft/OpenAPI.NET.OData/issues/582. We can always come back and do that for nullable types.

Reply:
Okay